### PR TITLE
make e2e framework as a separate module

### PR DESCRIPTION
**Pull request by chandrareddyp**
### What this PR does / why we need it
This PR is to make CLI E2E test framework and CLI Core E2E tests as independent go module. E2E framework user guide has updated for the plugin developers.
### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

1. PR E2E tests executed successfully
2. `make doc-lint` ran with no issues

### Release note

```release-note
CLI E2E framework is updated as a separate module
```

### CODE REVIEW

File: test/e2e/go.mod [hunk 1]
```diff
@@ -0,0 +1,39 @@
+module github.com/vmware-tanzu/tanzu-cli/test/e2e
+
+go 1.19
+
+replace github.com/vmware-tanzu/tanzu-cli => ../..
+
+require (
+	github.com/onsi/ginkgo v1.16.5
```
Comment (mpanchajanya) [inline test/e2e/go.mod:8]: @chandrareddyp ginkgo version is upgraded to  v2   in main. pls rebase and use ginkgo v2 here

**Review by mpanchajanya — CHANGES_REQUESTED** [review]
LGTM! update ginkgo to use v2

**Review by mpanchajanya — APPROVED** [review]
LGTM

**Review by anujc25 — APPROVED** [review]
LGTM.

**Pull request merged**